atomix/cluster/src/test/java/io/atomix/raft/ControllableRaftContexts.java

```diff
@@ -655,18 +660,21 @@ public void onWriteError(final Throwable error) {
     }
 
     @Override
-    public void onCommit(final IndexedRaftLogEntry indexed) {
-      final var entryChecksum = indexed.getPersistedRaftRecord().checksum();
-      final long index = indexed.index();
-      if (indexToChecksumMap.containsKey(index) && indexToChecksumMap.get(index) != entryChecksum) {
+    public void onCommit(final long index) {
+      if (committedIndexToChecksumMap.containsKey(index)
+          && pendingWriteToBeCommitted.containsKey(index)
+          && pendingWriteToBeCommitted.get(index).equals(committedIndexToChecksumMap.get(index))) {
```

Comment:
:x: Shouldn't the condition be inverted? It should be `!pendingWriteToBeCommitted.get(index).equals(committedIndexToChecksumMap.get(index))` right?

Reply:
ups :D I guess yes